src/Microsoft.Graph/Models/Extensions/ChangeNotificationEncryptedContent.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.IO;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Graph
+{
+    public partial class ChangeNotificationEncryptedContent
+    {
+        private static readonly Lazy<int> AESInitializationVectorSize = new Lazy<int>(() =>
+        {
+            using (AesCryptoServiceProvider provider = new AesCryptoServiceProvider())
+            {
+                return provider.LegalBlockSizes[0].MinSize;
+            }
+        });
+        /// <summary>
+        /// Validates the signature and decrypted content attached with the notification.
+        /// </summary>
+        /// <typeparam name="T">Type to deserialize the data to.</typeparam>
+        /// <param name="certificateProvider">Certificate provider to decrypt the content. The first parameter is the certificate ID provided when creating the subscription. The second is the certificate thumbprint. The certificate WILL be disposed at the end of decryption.</param>
+        /// <returns>Decrypted content as the provided type.</returns>
+        public async Task<T> Decrypt<T>(Func<string, string, Task<X509Certificate2>> certificateProvider)
+        {
+            return JsonConvert.DeserializeObject<T>(await Decrypt(certificateProvider));
```

Comment:
let me know if I should be using another deserialization method that is "more complete" and accounts for OData and such.

Reply:
The Serializer class makes sure that JsonConvert uses the right settings.  There is an ISerializer interface that sort of abstracts away the implementation.